neovim/msgpack_rpc/session.py

```diff
@@ -72,12 +73,19 @@ def request(self, method, *args, **kwargs):
         - Run the loop until the response is available
         - Put requests/notifications received while waiting into a queue
 
-        If the `async` flag is present and True, a asynchronous notification is
-        sent instead. This will never block, and the return value or error is
-        ignored.
+        If the `async_` flag is present and True, a asynchronous notification
+        is sent instead. This will never block, and the return value or error
+        is ignored.
         """
-        async = kwargs.pop('async', False)
-        if async:
+        if 'async_' not in kwargs and 'async' in kwargs:
+            warnings.warn(
+                '"async" attribute is deprecated. Use "async_" instead.',
+                DeprecationWarning,
+            )
```

Comment:
This looks like a good solution to me. 

If no one else comments, ping in a few days, then we merge.